### PR TITLE
fixed bug where tuner would not tune lr if also tuning batch_size

### LINKED ISSUE #4616: lr_tuner fails if called after tuning batch_size
https://github.com/PyTorchLightning/pytorch-lightning/blob/514cb22bd719e6ca056cacce730c8de875c9dbf6/pytorch_lightning/tuner/lr_finder.py#L411

The line above sets current_step to 2 if the batch_size tuner has already run.

https://github.com/PyTorchLightning/pytorch-lightning/blob/514cb22bd719e6ca056cacce730c8de875c9dbf6/pytorch_lightning/tuner/lr_finder.py#L419-L420

This causes best_loss to be zero and compare against the smoothed loss. This will cause the tuner to terminate and fail.

Also, just looks like this was already supposed to be changed.

### PR DESCRIPTION
**Pull request by Palzer**
## What does this PR do?

current_step should be set to the global_step instead of global_step + 1. This fixes that issue which was causing the tuner to not tune the learning rate if also tuning batch_size. The comment on the line also implies that this change should have already been made.

Fixes #4616

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Palzer** [comment]
Not sure about the failing timing test.

**Comment by SkafteNicki** [comment]
@Palzer the failing test seems to be unrelated to the change. Could you add some sort of test that shows the issue is solved?

**Review by Borda — APPROVED** [review]
lgtm, mind add some test got this case? :]

**Review by tchaton — COMMENTED** [review]
Great catch ! Mind adding a test ?

**Comment by Palzer** [comment]
My laptop died a week or so ago. New one should be arriving Monday and then I can take care of tests. Sorry for the delay!

**Comment by Borda** [comment]
@Palzer  there won't be any more releases in 1.1.x so I have changed your destination branch to feat 1.2
pls, there is no need to close this PR, just rebase on the target branch, or maybe easier just cherry-pick the commit (@kaushikb11 will help you if needed)

**Review by carmocca — APPROVED** [review]
LGTM

`self.beta**(current_step + 1)` avoids division by 0

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]
